### PR TITLE
dtc: build fdtput and fdtget

### PR DESCRIPTION
**Pull request by kszaq**
I cherry-picked the change from Raybuntu's LE fork. It adds building `fdtget` and `fdtput` tools useful to modify device trees in-place.

### CODE REVIEW

**Comment by lrusak** [comment]
Do we really need these tools in the build?

**Comment by kszaq** [comment]
If the reason to include `dtc` in RPi builds is to allow users modify device trees, then I would day `fdtget` and `fdtput` are useful as they provide a shortcut for modifications: no need to convert `dtb -> dts -> dtb`, you can modify `dtb` in-place.

**Comment by lrusak** [comment]
Why are we modifying dtb files in the build at all? Why is it not sufficient to do so at compile time?

**Comment by kszaq** [comment]
AFAIK we don't. It's a tool for users.

Out of curiosity, what is the purpose of having `dtc` for RPi?

**Comment by lrusak** [comment]
I believe the `dtoverlay` and `dtparam` tool need it IIRC

**Comment by adamg88** [comment]
This would certainly help with testing changes for vfd plus they are only a few Kb so no harm

**Comment by lrusak** [comment]
If you need it for testing purposes you can just copy the binaries to `/storage` and run them from there

**Comment by adamg88** [comment]
I can think of a number of use cases for these to benefit users ie users may want to change from meson-ir to amremote or to change LED configuration on boxes

**Comment by Raybuntu** [comment]
I agree. For C2 and LePotato we have config.ini to change dtb related settings. For S905, S912, WP2, WH this could be usable.

File: packages/tools/dtc/package.mk [hunk 1]
```diff
@@ -46,7 +46,15 @@ post_makeinstall_host() {
   touch $TOOLCHAIN/lib/$PKG_PYTHON_VERSION/site-packages/pylibfdt/__init__.py
 }
 
+pre_make_target() {
+  make clean
+}
+
 makeinstall_target() {
```
Comment (chewitt) [inline packages/tools/dtc/package.mk]: I'd suggest making the new additions $PROJECT based so we're not adding them to images where they're not really relevant, e.g.

```
  makeinstall_target() {
    mkdir -p $INSTALL/usr/bin
      cp -P $PKG_BUILD/dtc $INSTALL/usr/bin

    if [ "$PROJECT" = "Amlogic" ]; then
      mkdir -p $INSTALL/usr/lib
        cp -P $PKG_BUILD/fdtput $INSTALL/usr/bin/
        cp -P $PKG_BUILD/fdtget $INSTALL/usr/bin/
        cp -P $PKG_BUILD/libfdt/libfdt.so $INSTALL/usr/lib/
    fi
```

**Comment by MilhouseVH** [comment]
Building `dtc` with this PR for RPi and RPi2 fails for me: http://ix.io/KMZ

`clean` is defined as:
```
clean: libfdt_clean pylibfdt_clean tests_clean
        @$(VECHO) CLEAN
        rm -f $(STD_CLEANFILES)
        rm -f $(VERSION_FILE)
        rm -f $(BIN)
        rm -f dtc-*.tar dtc-*.tar.sign dtc-*.tar.asc
```
However in my build scripts, I have defined `BIN=/home/neil/projects` which is most likely the cause of:
```
rm: cannot remove '/home/neil/projects': Is a directory
```
Thankfully this wasn't `rm -f $(BIN)`...

This looks like an error in the Makefile, or we need to specify an alternative (and safe) value for BIN (ie. `BIN=""`) as the initial value of BIN is not defined within the Makefile.

**Comment by MilhouseVH** [comment]
```
    make clean BIN=
```
will avoid the issue of `dtc` trying to delete parts of my file system.

There's also a new version 1.4.6 available (no BIN-related changes).

**Comment by MilhouseVH** [comment]
@kszaq can you add `BIN=` to the end of `make clean`?

**Comment by kszaq** [comment]
@MilhouseVH Done.

**Pull request merged**